src/SumoCoders/FrameworkCoreBundle/Mail/MessageFactory.php

```diff
@@ -0,0 +1,253 @@
+<?php
+
+namespace SumoCoders\FrameworkCoreBundle\Mail;
+
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use TijsVerkoyen\CssToInlineStyles\CssToInlineStyles;
+
+class MessageFactory
+{
+    /**
+     * @var array
+     */
+    protected $sender = array();
+
+    /**
+     * @var array
+     */
+    protected $replyTo = array();
+
+    /**
+     * @var array
+     */
+    protected $to = array();
+
+    /**
+     * @var EngineInterface
+     */
+    protected $template;
+
+    /**
+     * @var string
+     */
+    protected $templatePath;
+
+    /**
+     * @var string
+     */
+    protected $cssPath;
+
+    /**
+     * MessageFactory constructor.
+     *
+     * @param EngineInterface $template
+     */
+    public function __construct(EngineInterface $template, $templatePath, $cssPath)
+    {
+        $this->template = $template;
+        $this->templatePath = $templatePath;
+        $this->cssPath = $cssPath;
+    }
+
+    /**
+     * Set the default sender
+     *
+     * @param string      $email
+     * @param string|null $name
+     */
+    public function setDefaultSender($email, $name = null)
+    {
+        if ($name !== null) {
+            $this->sender = array($email => $name);
+        } else {
+            $this->sender = $email;
+        }
+    }
+
+    /**
+     * Set the default reply-to
+     *
+     * @param string      $email
+     * @param string|null $name
+     */
+    public function setDefaultReplyTo($email, $name = null)
+    {
+        if ($name !== null) {
+            $this->replyTo = array($email => $name);
+        } else {
+            $this->replyTo = $email;
+        }
+    }
+
+    /**
+     * Set the default to
+     *
+     * @param string      $email
+     * @param string|null $name
+     */
+    public function setDefaultTo($email, $name = null)
+    {
+        if ($name !== null) {
+            $this->to = array($email => $name);
+        } else {
+            $this->to = $email;
+        }
+    }
+
+    /**
+     * Create a message
+     *
+     * @param string|null $subject
+     * @param string|null $html
+     * @param string|null $alternative
+     * @return \Swift_Message
+     */
+    protected function createMessage($subject = null, $html = null, $alternative = null)
+    {
+        $message = $this->createDefaultMessage();
+
+        if ($subject != '') {
+            $message->setSubject($subject);
+        }
+
+        if ($html != '' && $alternative != '') {
+            $message->setBody(
+                $this->wrapInTemplate($html),
+                'text/html'
+            );
+            $message->addPart($alternative, 'text/plain');
+        } elseif ($html != '' && $alternative == '') {
```

Comment:
I think this could be a lot cleaner if you used early returns instead of elseif statements.